src/k8s-engine/lima.ts

```diff
@@ -241,6 +237,21 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     return this.#desiredPort;
   }
 
+  protected async ensureArchitectureMatch() {
+    if (os.platform().startsWith('darwin')) {
+      const fileArch = this.arch === 'aarch64' ? 'arm64' : this.arch;
+      const { stdout } = await childProcess.spawnFile(
+        'file', [this.limactl],
+        { stdio: ['inherit', 'pipe', console] });
+
+      if (!stdout.trim().match(`executable ${ fileArch }$`)) {
+        const otherArch = { aarch64: 'x86_64', x86_64: 'aarch64' }[this.arch];
```

Comment:
Alternatively, `const otherArch = stdout.trim().split().pop();` (basically).

Reply:
It should work (because it is the last word); normally I worry about `split` no doing the right thing when there are double spaces or whatever.

I kind of think the current version is fine as-is.

Reply:
Yeah, I think it's fine as-is.